ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,274 @@
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        time_flag = False
+        while datetime.datetime.now() < endtime:
+            mgr_count = 0
+            for node in mgr_node_list:
+                if self.check_mgr_exists_on_host(host=node):
+                    log.info(f"Mgr {node} is up and running in the cluster.")
+                    mgr_count = mgr_count + 1
+                    if new_mgr_count == mgr_count:
+                        log.info(" All mgr nodes to  up and running")
+                        time_flag = True
+                        break
+                else:
+                    log.info(f"Mgr {node} is not running in the cluster.")
+                    time.sleep(5)
+            if time_flag:
+                log.info(f"All mgrs are running and not checking up to {endtime}")
+                break
+        if mgr_count != new_mgr_count:
+            log.info(f" The {mgr_node_list} are not up and running")
+            return False
+        return True
+
+    def check_mgr_exists_on_host(self, host) -> bool:
+        """
+        Method to check if the mgr daemon exists on the given host
+        Args:
+            host: name of the host where existence of mgr needs to be checked
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        cmd = f"ceph orch ps {host}"
+        out = self.rados_obj.run_ceph_command(cmd=cmd, client_exec=True)
+        for entry in out:
+            if entry["daemon_type"] == "mgr":
+                log.debug("mgr daemon present on the host")
+                return True
+        log.debug("mgr daemon not present on the host")
+        return False
+
+    def get_mgr_stats(self):
+        """
+        Method to retrive the mgr daemon stats
+        Args: none
+        Returns:
+             mgr daemon stats in json format.
+        """
+        cmd = "ceph mgr stat"
+        mgr_stats = self.rados_obj.run_ceph_command(cmd)
+        return mgr_stats
+
+    def set_mgr_fail(self, host):
+        """
+        Method to fail the mgr host
+        Args:
+         host : mgr host name
+        Return:
+             Return the output of the execution of the command
+        """
+        cmd = f"ceph mgr fail {host}"
+        out_put = self.rados_obj.run_ceph_command(cmd)
+        time.sleep(10)
+        return out_put
+
+    def get_mgr_daemon_list(self):
+        """
+        Method to retrive the mgr daemon list not the host name.
+        For example the  mgr daemon name like
+         - ceph-rados-tfa-otrj8b-node1-installer.agbrji, ceph-rados-tfa-otrj8b-node2.yplzgj
+        Args:
+             none
+        Return: Mgr active and standby daemon list.
+        """
+        mgr_list = []
+        out_put = self.rados_obj.run_ceph_command(cmd="ceph mgr dump")
+        mgr_list.append(out_put["active_name"])
+        for standby_mgr in out_put["standbys"]:
+            mgr_list.append(standby_mgr["name"])
+        print(mgr_list)
```

Comment:
please use logger to log details into file.

Reply:
Added the logger details.